backend/typescript/services/implementations/EntityServicePg.ts

```diff
@@ -0,0 +1,130 @@
+            if (!entity) {
+                throw new Error(`Entity id ${id} not found`);
+            }
+        } catch (error) {
+            Logger.error(`Failed to get entity. Reason = ${error.message}`);
+            throw error;
+        }
+
+        return {
+            id: entity.id,
+            stringField: entity.string_field,
+            intField: entity.int_field,
+            enumField: entity.enum_field,
+            stringArrayField: entity.string_array_field,
+            boolField: entity.bool_field,
+        };
+    }
+
+    async getEntities(): Promise<EntityResponseDTO[]> {
+        try {
+            const entities: Array<PgEntity> = await PgEntity.findAll({ raw: true });
+            return entities.map((entity) => ({
+                id: entity.id,
+                stringField: entity.string_field,
+                intField: entity.int_field,
+                enumField: entity.enum_field,
+                stringArrayField: entity.string_array_field,
+                boolField: entity.bool_field,
+            }));
+        } catch (error) {
+            Logger.error(`Failed to get entities. Reason = ${error.message}`);
+            throw error;
+        }
+    }
+
+    async createEntity(entity: EntityRequestDTO): Promise<EntityResponseDTO> {
+        let newEntity: PgEntity | null;
+        try {
+            newEntity = await PgEntity.create({
+                string_field: entity.stringField,
+                int_field: entity.intField,
+                enum_field: entity.enumField,
+                string_array_field: entity.stringArrayField,
+                bool_field: entity.boolField,
+            });
+        } catch (error) {
+            Logger.error(`Failed to create entity. Reason = ${error.message}`);
+            throw error;
+        }
+        return {
+            id: newEntity.id,
+            stringField: newEntity.string_field,
+            intField: newEntity.int_field,
+            enumField: newEntity.enum_field,
+            stringArrayField: newEntity.string_array_field,
+            boolField: newEntity.bool_field,
+        };
+    }
+
+    async updateEntity(
+        id: string,
+        entity: EntityRequestDTO,
+    ): Promise<EntityResponseDTO | null> {
+        let resultingEntity: PgEntity | null;
+        let updatedEntity: [number, PgEntity[]] | null;
+        try {
+            updatedEntity = await PgEntity.update(
+                {
+                    string_field: entity.stringField,
+                    int_field: entity.intField,
+                    enum_field: entity.enumField,
+                    string_array_field: entity.stringArrayField,
+                    bool_field: entity.boolField,
+                },
+                { where: { id }, returning: true },
+            );
+
+            if (!updatedEntity) {
+                throw new Error(`Entity id ${id} not found`);
+            }
+            resultingEntity = updatedEntity[1][0];
```

Comment:
Should do an error check on `updatedEntity[0]` first (the count of updated entities), otherwise `updatedEntity[1][0]` might throw array out of bounds error

Also linter is suggesting array destructuring syntax
```
[, [resultingEntity]] = updatedEntity;
```